### PR TITLE
Control fathomlondon.com and fathom-newyork.com from a single repository

### PR DESCRIPTION
**Pull request by andrewjdick**
## Ticket

_Which GitHub issue does your code resolve?_

This implementation would allow us to control both fathomlondon.com and fathom-newyork.com from a single repository. 

**Netlify:**
We can set the _Production Branch_ on each site's Netlify deploy settings, i.e: 

fathomlondon: `master`
fathom-newyork: `new-york` 

We also add a `GATSBY_REGION` variable to each site's Build environment variables:
 
image

image

**Local Testing:**
For local testing, the `GATSBY_REGION` env variable is set the start scripts:

```
"start": "yarn start:london",
"start:london": "GATSBY_REGION=london yarn develop",
"start:newYork": "GATSBY_REGION=newYork yarn develop",
```

As well as determining the site's metadata, the env variable needs to be accessed on the client-side. We need this as the site copy on `/contact` changes based on the region. 

https://www.gatsbyjs.org/docs/environment-variables/#client-side-javascript

## Proposed Changes

_What changes does your code introduce?_

- 🔧Configured  `GATSBY_REGION` env variable to push latest changes to both sites.

## Types of changes

_What types of changes does your code introduce?_

- [ ] 🔥 Breaking change (fix or feature that would cause existing functionality to change)
- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [ ] 🎨 Improve general format/structure
- [ ] ⬆️ Upgrading dependencies
- [x] 🔧 Configuration

### CODE REVIEW

**Comment by peduarte** [comment]
oh that's nice :) 

Not about `GATSBY_REGION`? This it not exclusively a Gatsby thing, right?

Maybe we can use more friendly names even, like `OFFICE`? 

Can you name the env variables in uppercase? `LONDON / NEW_YORK`?

@ibrennan @artisologic @matthewfowles  what do you think?

**Comment by andrewjdick** [comment]
@peduarte 

`GATSBY_REGION` => `GATSBY_OFFICE`✅
`london` => `LONDON` ✅

> Not about GATSBY_REGION? This it not exclusively a Gatsby thing, right?

This is an interesting one. I added the `GATSBY_` prefix because we need to be able to access the env variable when we're testing locally. I tried the below, but it returned as `undefined`:

- creating an `env.development.london` and a `env.development.newyork` file, each with their own respective `OFFICE` variable.
- Specifying the ENVFILE in the start scripts, i.e. `"start:london": ""ENVFILE=.env.development.london yarn develop",` 

I ended up using an **OS Env Var** from the [Gatsby Docs](https://www.gatsbyjs.org/docs/environment-variables/#client-side-javascript):

> In addition to these Project Environment Variables defined in .env.* files, you could also define OS Env Vars. OS Env Vars which are prefixed with GATSBY_ will become available in browser JavaScript.

**Comment by peduarte** [comment]
Does it only work if the variable name starts with `GATSBY_`?

File: src/pages/contact.js [hunk 1]
```diff
@@ -8,7 +8,13 @@ import { ContactCard } from '../components/ContactCard';
...
 
 function Contact({ data }) {
-	const { tel, email } = data.site.siteMetadata;
+	const {
+		tel,
+		email,
+		address: { street, city, zip },
+	} = data.site.siteMetadata;
+
+	const { GATSBY_REGION } = process.env;
```
Comment (peduarte) [inline src/pages/contact.js]: let's import this outside of the function scope

**Comment by peduarte** [comment]
Oh ok, so what you're saying is that it needs to start with `GATSBY_` so you have access from it at the component level? 

I think maybe this should be done with 
- `.env`: gets used in dev
- `.env.production`: gets used in prod

Rather than on Netlify UI.

File: gatsby-config.js [hunk 2]
```diff
@@ -3,17 +3,41 @@ const {
...
 	NETLIFY_ENV = NODE_ENV,
+	GATSBY_REGION,
 } = process.env;
 
 const isNetlifyProduction = NETLIFY_ENV === 'production';
 const siteUrl = isNetlifyProduction ? NETLIFY_SITE_URL : NETLIFY_DEPLOY_URL;
 
-module.exports = {
-	siteMetadata: {
+console.log({ GATSBY_REGION });
```
Comment (artisologic) [inline gatsby-config.js]: Do we want to leave this in?

**Comment by artisologic** [comment]
Yeah I agree it's a bit confusing with the `GATSBY_` prefix but it's quite similar to the `REACT_APP_` prefix with create-react-app for example so if it's nothing we can do about then I get it.

I think the "office" naming is clearer though, better than region.

File: gatsby-config.js [hunk 3]
```diff
@@ -3,17 +3,41 @@ const {
...
+
+const regionalMetadata = {
```
Comment (artisologic) [inline gatsby-config.js]: if we go with the office naming, then this can be `officesMetadata`

File: gatsby-config.js [hunk 4]
```diff
@@ -3,17 +3,41 @@ const {
...
+			city: 'New York',
+			zip: 'NY 10001',
+		},
+	},
+};
+
+module.exports = {
+	siteMetadata: {
+		siteUrl,
+		...regionalMetadata[GATSBY_REGION],
```
Comment (peduarte) [inline gatsby-config.js]: Also im not sure I like this dynamic approach of selecting the right data based off of the env var. It feels too error prone and coupled together.

I'd prefer to be more explicit here.

```js
const london = {...};
const newYork = {...};
const region = GATSBY_REGION === 'LONDON' ? london : newYork; 

module.exports = {
	siteMetadata: {
		siteUrl,
		...region,

```

File: src/layouts/index.js [hunk 5]
```diff
@@ -29,12 +29,21 @@ class Layout extends React.Component {
...
 							}
 						}
 					}
 				`}
 				render={({ site: { siteMetadata } }) => {
-					const { email, tel } = siteMetadata;
+					const {
+						email,
+						tel,
+						address: { street, city, zip },
```
Comment (artisologic) [inline src/layouts/index.js]: I'm not a fan of nested destructuring, I feel it doesn't help with readability.
How about doing something like:

```jsx
const { email, tel, address } = siteMetadata;
const { street, city, zip } = address;
```

File: src/pages/contact.js [hunk 6]
```diff
@@ -8,7 +8,13 @@ import { ContactCard } from '../components/ContactCard';
 import { Nav } from '../components/Nav';
 
 function Contact({ data }) {
-	const { tel, email } = data.site.siteMetadata;
+	const {
+		tel,
+		email,
+		address: { street, city, zip },
```
Comment (artisologic) [inline src/pages/contact.js]: same comment here

File: gatsby-config.js (hunk 2 above)
Comment (andrewjdick) [reply gatsby-config.js]: Nope. I just left that in for now for testing on the site deploys. 

I'll remove it once the other changes are in 👍

File: src/pages/index.js [hunk 7]
```diff
@@ -28,7 +28,11 @@ class Index extends React.Component {
 	}
 
 	render() {
-		const { tel, email } = this.props.data.site.siteMetadata;
+		const {
+			tel,
+			email,
+			address: { street, city, zip },
```
Comment (artisologic) [inline src/pages/index.js]: same thing

**Comment by andrewjdick** [comment]
@peduarte 

> Rather than on Netlify UI.

The problem is we need some way of determining the office when pushing to either site. 

The advantage of adding it on the Netlify UI is that each site controls its own office. 
I'm not sure how you would do a conditional within an `.env` file to determine the office?

If I push to the new-york branch, how will the env file know to assign newyork in the OFFICE variable?

File: gatsby-config.js (hunk 4 above)
Comment (andrewjdick) [reply gatsby-config.js]: Agreed. Should we fall back to London as the default?

**Comment by peduarte** [comment]
@andrewjdick oh yeah makes sense

File: gatsby-config.js (hunk 4 above)
Comment (peduarte) [reply gatsby-config.js]: Yeah

**Comment by andrewjdick** [comment]
@peduarte @artisologic I've actioned all requested changes. 

Are you happy for me to go ahead with this?

**Pull request merged**